### ISSUE TITLE
Importing a .stylex.ts does not work with tsconfig paths settings

### ISSUE THREAD

**Issue by ethanjdiamond** [facebook/stylex#40]
**The problem**
When importing a variable file like is listed in [the docs](https://stylexjs.com/docs/learn/theming/defining-variables/), the build is only succeeding if the import path is a relative path (i.e. ../src/lib/theme/theme.stylex), but fails if the import path uses the tsconfig "paths" setting (i.e. @/lib/theme/theme.stylex when tsconfig paths is { '@/*' : '.src/*'} ). 

With non-relative paths, it gives the `Only static values are allowed inside of a stylex.create() call` error message.

**How to reproduce**

Steps to reproduce:
1. Create a next.js app with the default of `paths: { '@/*' : '.src/*'}` in the tsconfig.json file.
2. Create a file called theme.stylex.ts in `/src/lib/` that defines a color using `export const colors = stylex.defineVars({test: '#000000'})`
3. Try to import the colors using `import { colors } from @/lib/theme.stylex` and use `colors.test` in a `stylex.create` call.

**Expected behavior**
I would expect this to allow `colors` to be used in `stylex.create` as specified in [the docs](https://stylexjs.com/docs/learn/theming/defining-variables/).

**Environment (include versions). Did this work in previous versions?**
- next 14.0.2 using an app router
- @stylexjs/stylex - 0.3.0
- @stylexjs/nextjs-plugin - 0.3.0

**Comment by rayan1810** [comment]
If this support is not deliberately excluded, I would like to work on a PR to add this feature.

**Comment by nmn** [comment]
The StyleX Babel plugin relies on the `require.resolve` function to determine the absolute file path of the module being imported. If aliases are setup in a way that Node would understand, they would work. e.g. mono-repos.

@rayan1810 Happy for you to solve this issue. Please start by describing an outline of how you would solve it.

**Comment by nmn** [comment]
@rayan1810 This solution feels too specific to Next and TS. Perhaps there is a way to pass in an alias configuration to the Babel plugin itself? Then, the Next.js plugin can read the aliases and inject them into the Babel plugin while using it.

**Comment by AbhiPartha** [comment]
This issue seems to even happen for relative imports

For example you can use this repo to reproduce it. 
https://github.com/millnut/storybook_foundations/tree/feature/stylex-support

Any idea why this could be happening ?
Node version: v18.18.2

**Comment by nmn** [comment]
A quick update is that StyleX can now be configured to resolve aliases, although the configuration requires absolute paths. See the Next example in the repo for an example.

@AbhiPartha Please don't post the same comment in multiple issues. I'll keep this one as it's on an open issue and delete the other one for now.

**Comment by nmn** [comment]
@AbhiPartha yes. this is a known issue. The StyleX babel plugin can be configured to understand aliases. We don't automatically discover and read aliases defined in a `tsconfig` file.

**Comment by jderboven** [comment]
Any clues on how we could do it for a nx monorepo project with a lib containing styleX ? 
I would like to make a shareable theming lib within my monorepo with variables and styles defined with stylex so I can be able to share tokens to all of my apps. 

Actually getting the same issue but there are no relative paths here.

**Comment by nmn** [comment]
@jderboven Depends on the framework/bundler you're using. AFAIK, using `nx` should not change how it is configured. 

Generally, you need to configure the build for your app to also compile the packages you're importing. These external packages may be from your mono-repo or `node_modules`. In Next.js, for example, you need to add the package names to [`transpilePackages`](https://nextjs.org/docs/app/api-reference/next-config-js/transpilePackages) key.

Vite has a similar option AFAIK. (Although, you'd need to delete the Vite cache between builds for things to work consistently.

> Actually getting the same issue but there are no relative paths here.

Could you explain what this means?

**Comment by corydeppen** [comment]
I'm running into this issue using the latest version of @stylexjs/unplugin (0.17.4), where I get the following error despite the module being named as expected (`vars.stylex.ts`):

```
Could not resolve the path to the imported file. Please ensure that the theme file has a .stylex.js or .stylex.ts extension and follows the rules for defining variables.
```

It's been a while since the last response, so I was wondering if there are plans to include this in the plugin or if there's a recommended workaround for using aliases with Vite - hopefully without adding Babel config if possible. It's worth noting non-StyleX modules can be imported via aliases as expected across the app.

**Comment by nmn** [comment]
We have an open PR for this. I'll rebase it soon and get it merged. In the meantime, you'll have to rely on configuring the StyleX babel plugin with aliases manually.

**Comment by RafadanaM** [comment]
Just putting this out there. I am trying tanstack start with bun on windows and I keep getting the same `Could not resolve the path to the imported file...` error even though I already provided `aliases` option in vite config.

I had to change the alias to something like:
```javascript
aliases: {
  "@/*": [new URL("./src/", import.meta.url).href + "*"], // resolves to file:///[drive_code]:/[path/to/project]/src/*
},
```

Might help some people who's confused about this.
